src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static partial class OperationExtensions
+    {
+        /// <summary>
+        /// Returns the <see cref="ValueUsageInfo"/> for the given operation.
+        /// This extension can be removed once https://github.com/dotnet/roslyn/issues/25057 is implemented.
+        /// </summary>
+        public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
+        {
+            /*
+            |    code         | Read | Write | ReadableRef | WritableRef |
+            | nameof(x)       |      |       |             |             |
+            | x.Prop = 1      |      |  ✔️   |             |             |
+            | x.Prop += 1     |  ✔️  |  ✔️   |             |             |
+            | x.Prop++        |  ✔️  |  ✔️   |             |             |
+            | Foo(x.Prop)     |  ✔️  |       |             |             |
+            | Foo(x.Prop)*    |      |       |     ✔️      |             |
+            | Foo(out x.Prop) |      |       |             |     ✔️      |
+            | Foo(ref x.Prop) |      |       |     ✔️      |     ✔️      |
```

Comment:
there is code in Find-Refs to compute this to determine how a property is being used.  Can you update that code as well to use this facility.  We probably also want to update the IsWrittenTo and IsOnlyWrittenTo helpers.  This will help prove out that this code is correct.

Reply:
I will file a follow-up issue to take this up next week and separate it from this PR.

---
In reply to: [212735745](https://github.com/dotnet/roslyn/pull/29511#discussion_r212735745) [](ancestors = 212735745)

Reply:
Tracked in [https://github.com/dotnet/roslyn/issues/29519](https://github.com/dotnet/roslyn/issues/29519)

---
In reply to: [212781438](https://github.com/dotnet/roslyn/pull/29511#discussion_r212781438) [](ancestors = 212781438,212735745)